ammico/prompt_builder.py

```diff
@@ -0,0 +1,287 @@
+                - If answer is found in visual information → use that
+                - ONLY respond "Cannot be determined" if truly no information exists anywhere
+
+        Return ONLY this format:
+
+        VQA Answers:
+        1. <answer to question 1>
+        2. <answer to question 2>
+        [etc.]"""
+
+        return str_to_return
+
+    @staticmethod
+    def questions_module(questions: List[str]) -> str:
+        """Format questions list."""
+        questions_text = "\n".join(
+            [f"{i + 1}. {q.strip()}" for i, q in enumerate(questions)]
+        )
+        str_to_return = f"""## Questions to Answer
+
+        {questions_text}
+        """
+        return str_to_return
+
+    @staticmethod
+    def vqa_context_module(vqa_bullets: List[str], is_final: bool = False) -> str:
+        """VQA context (frame-level or clip-level answers)."""
+        if is_final:
+            header = """## SEGMENT-Level Answer Context (Reference Only)
+
+            The following are answers to above questions obtained for each segment of the video.
+            These answers are associated with the timestamp of each segment's beginning:"""
+        else:
+            header = """## FRAME-Level Answer Context (Reference Only)
+
+            For each question, the following are frame-level answers provided as reference.
+            If these answers are "Cannot be determined", do not accept that as final—instead, 
+            use visual and audio information to answer the question:"""
+
+        bullets_text = "\n".join(vqa_bullets)
+        return f"{header}\n\n{bullets_text}"
+
+    @classmethod
+    def build_frame_prompt(
+        cls, include_vqa: bool = False, questions: Optional[List[str]] = None
+    ) -> str:
+        """Build prompt for frame-level analysis."""
+        modules = [cls.ROLE_MODULE, cls.visual_frames_module()]
+
+        if include_vqa and not questions:
+            raise ValueError("Questions must be provided when VQA should be included.")
+
+        if include_vqa:
+            modules.append(cls.summary_vqa_task(ProcessingLevel.FRAME))
+            modules.append(cls.questions_module(questions))
+        else:
+            modules.append(cls.summary_task())
+
+        modules.append(cls.CONSTRAINTS_MODULE)
+        return "\n\n".join(modules)
+
+    @classmethod
+    def build_clip_prompt(
+        cls,
+        frame_bullets: List[str],
+        include_audio: bool = False,
+        audio_transcription: Optional[List[Dict]] = None,
+        include_vqa: bool = False,
+        questions: Optional[List[str]] = None,
+        vqa_bullets: Optional[List[str]] = None,
+    ) -> str:
+        """Build prompt for clip-level analysis."""
+        modules = [cls.ROLE_MODULE, cls.visual_captions_module(frame_bullets)]
+
+        if include_audio and audio_transcription:
+            modules.append(cls.audio_module(audio_transcription))
+
+        if include_vqa and not questions:
```

Comment:
Here I have added the same check as above. If there are no questions provided, raise an error. Again I am wondering why this does both summary and VQA.